From 5cb8340bf9502e1b6d4c321f4b508dbdc3a9c113 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:51:11 -0400
Subject: [fs] nfsd: use session limits to release send buffer reservation

Message-id: <1403560314-32525-130-git-send-email-bfields@redhat.com>
Patchwork-id: 84499
O-Subject: [Patch RHEL7 129/172] nfsd4: use session limits to release send buffer reservation
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

Once we know the limits the session places on the size of the rpc, we
can also use that information to release any unnecessary reserved reply
buffer space.

Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 32aaa62ede574ff99b020b4ee3ff6f9cfc9f0099
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index 612b85a..62b882d 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -2290,6 +2290,7 @@ nfsd4_sequence(struct svc_rqst *rqstp,
         nfserr_rep_too_big;
  if (xdr_restrict_buflen(xdr, buflen - 2 * RPC_MAX_AUTH_SIZE))
   goto out_put_session;
+ svc_reserve(rqstp, buflen);
 
  status = nfs_ok;
  /* Success! bump slot seqid */
-- 
1.7.1